ouroboros-network/src/Ouroboros/Network/Protocol/LocalTxSubmission/Type.hs

```diff
@@ -45,6 +45,13 @@ data LocalTxSubmission tx reject where
   StDone   :: LocalTxSubmission tx reject
 
 
+-- | Isomorphic with Maybe but with a name that better describes its purpose and
+-- usage.
+data SubmitResult reason
```

Comment:
@erikd I think it makes more sense if we define this type in `Client`.  This is client related only type, rather than a part of the protocol.

Reply:
`Server` also needs this type.